backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java

```diff
@@ -2013,4 +2024,37 @@ private ICompositeQueryFilter<I_C_Invoice_Candidate> toFilter(@NonNull final Inv
 
 		return filter;
 	}
+
+	@NonNull
+	private IQueryBuilder<I_C_Invoice_Line_Alloc> retrieveInvoiceLineAllocQueryBuilder(@NonNull final InOutId inOutId)
+	{
+		final IQuery<I_C_InvoiceCandidate_InOutLine> candidateInOutLine = retrieveCandidateInOutLineQuery(inOutId);
+
+		return queryBL.createQueryBuilder(I_C_Invoice_Line_Alloc.class)
+				.addOnlyActiveRecordsFilter()
+				.addInSubQueryFilter(I_C_Invoice_Line_Alloc.COLUMNNAME_C_Invoice_Candidate_ID, I_C_InvoiceCandidate_InOutLine.COLUMNNAME_C_Invoice_Candidate_ID, candidateInOutLine);
+	}
+
+	@NonNull
+	private IQuery<I_C_InvoiceCandidate_InOutLine> retrieveCandidateInOutLineQuery(@NonNull final InOutId inOutId)
+	{
+		return queryBL.createQueryBuilder(I_M_InOutLine.class)
+				.addEqualsFilter(I_M_InOutLine.COLUMNNAME_M_InOut_ID, inOutId)
+				.addOnlyActiveRecordsFilter()
+				.andCollectChildren(I_C_InvoiceCandidate_InOutLine.COLUMNNAME_M_InOutLine_ID, I_C_InvoiceCandidate_InOutLine.class)
+				.create();
+	}
+
+	private boolean isAggregationWithAttributeAggregatePerInOut(@NonNull final IQueryBuilder<I_C_Invoice_Line_Alloc> invoiceLineAllocQueryBuilder)
+	{
+		return invoiceLineAllocQueryBuilder
+				.andCollect(I_C_Invoice_Candidate.COLUMNNAME_C_Invoice_Candidate_ID, I_C_Invoice_Candidate.class)
+				.andCollect(I_C_Invoice_Candidate.COLUMNNAME_HeaderAggregationKeyBuilder_ID, I_C_Aggregation.class)
+				.andCollectChildren(I_C_AggregationItem.COLUMNNAME_C_Aggregation_ID, I_C_AggregationItem.class)
+				.addEqualsFilter(I_C_AggregationItem.COLUMNNAME_Type, X_C_AggregationItem.TYPE_Attribute)
+				.andCollect(I_C_AggregationItem.COLUMNNAME_C_Aggregation_Attribute_ID, I_C_Aggregation_Attribute.class)
+				.addEqualsFilter(I_C_Aggregation_Attribute.COLUMNNAME_Code, ATTRIBUTE_CODE_AggregatePer_M_InOut_ID)
```

Comment:
from a quick glimpse IDK where exactly, but i think there have to be filters for IsActive, because the propery might be there, but not deactivated

Reply:
Changed the implementation to use `M_MatchInv`